src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java

```diff
@@ -342,964 +375,698 @@ public sealed interface Subtask<T> extends Supplier<T> permits SubtaskImpl {
          *
-         * <p> Construction captures the current thread's {@linkplain ScopedValue scoped
-         * value} bindings for inheritance by threads started in the task scope. The
-         * {@linkplain StructuredTaskScope##TreeStructure Tree Structure} section in the class description
-         * details how parent-child relations are established implicitly for the purpose
-         * of inheritance of scoped value bindings.
+         * @apiNote The thread factory will typically create
+         * <a href="{@docRoot}/java.base/java/lang/Thread.html#virtual-threads">virtual threads</a>,
+         * maybe with names for monitoring purposes, an {@linkplain Thread.UncaughtExceptionHandler
+         * uncaught exception handler}, or other properties configured.
          *
-         * @param name the name of the task scope, can be null
-         * @param factory the thread factory
+         * @see #fork(Callable)
          */
-        public ShutdownOnFailure(String name, ThreadFactory factory) {
-            super(name, factory);
-        }
+        Configuration withThreadFactory(ThreadFactory threadFactory);
 
         /**
-         * Constructs a new unnamed {@code ShutdownOnFailure} that creates virtual threads.
-         *
-         * @implSpec This constructor is equivalent to invoking the 2-arg constructor with
-         * a name of {@code null} and a thread factory that creates virtual threads.
+         * {@return a new {@code Configuration} object with the given name}
+         * The other components are the same as this object. A scope is optionally
+         * named for the purposes of monitoring and management.
+         * @param name the name
          */
-        public ShutdownOnFailure() {
-            this(null, Thread.ofVirtual().factory());
-        }
-
-        @Override
-        protected void handleComplete(Subtask<?> subtask) {
-            if (subtask.state() == Subtask.State.FAILED
-                    && firstException == null
-                    && FIRST_EXCEPTION.compareAndSet(this, null, subtask.exception())) {
-                super.shutdown();
-            }
-        }
+        Configuration withName(String name);
 
         /**
-         * Wait for all subtasks started in this task scope to complete or for a subtask
-         * to {@linkplain Subtask.State#FAILED fail}.
-         *
-         * <p> This method waits for all subtasks by waiting for all threads {@linkplain
-         * #fork(Callable) started} in this task scope to finish execution. It stops waiting
-         * when all threads finish, a subtask fails, or the current thread is {@linkplain
-         * Thread#interrupt() interrupted}. It also stops waiting if the {@link #shutdown()
-         * shutdown} method is invoked directly to shut down this task scope.
+         * {@return a new {@code Configuration} object with the given timeout}
+         * The other components are the same as this object.
```

Comment:
```suggestion
         * The other other configuration options are preserved.
```

Reply:
The class introduces them as components of the configuration rather than "options" so I think I'll leave this as is.